src/allmydata/test/test_crypto.py

```diff
@@ -0,0 +1,272 @@
+import six
+import unittest
+
+from base64 import b64decode
+from binascii import a2b_hex, b2a_hex
+from os import path
+
+from allmydata.crypto.aes import AES
+from allmydata.crypto import ed25519, rsa
+
+RESOURCE_DIR = path.join(path.dirname(__file__), 'data')
+
+
+class TestRegression(unittest.TestCase):
+    '''
+    These tests are regression tests to ensure that the upgrade from `pycryptopp` to `cryptography`
+    doesn't break anything. They check that data encrypted with old keys can be decrypted with new
+    keys.
+    '''
+
+    AES_KEY = b'My\x9c\xc0f\xd3\x03\x9a1\x8f\xbd\x17W_\x1f2'
+    IV = b'\x96\x1c\xa0\xbcUj\x89\xc1\x85J\x1f\xeb=\x17\x04\xca'
+
+    with open(path.join(RESOURCE_DIR, 'pycryptopp-rsa-2048-priv.txt')) as f:
+        '''
```

Comment:
I think this should use `pkg_resources` instead (in case, e.g., it's being run from a tarball). Approximately `pkg_resources.resource_stream("allmydata", "test/data/pycryptopp-rsa-2048-priv.txt")`

Reply:
If this is a goal there needs to be some automated testing that verifies it.

Reply:
We're shipping package-data as per the setup, so I assume the goal is "when we build this with setup.py, the data is there"...? (In which case, isn't `pkg_resources` "the way" you're supposed to get package data?)

Reply:
I guess my idea is that the code in this PR looks like it works.  If it's not what we want, there should be an automated test that makes it look like it doesn't work.  Or maybe the broader idea is that minimizing folk lore about contribution is a good thing.

The automated test that makes this look like it doesn't work would be ...a CI configuration that makes ... a Tahoe-LAFS egg?  and then runs the test suite against that.  I guess?  Maybe an egg isn't the right thing ... I don't know.

Do we have a good idea of where/when `pkg_resources.resource_stream(...)` works and this `open(...)` doesn't?

Reply:
I might be guilty of cargo-culting here, because I don't know for sure when `open(..)` doesn't work .. just "following what people advise" regarding `pkg_resources`. I guess if it works for local `tox` and works in CI, then "it works" until someone files a bug?